### PR TITLE
Quick menu: shape quick actions

### LINKED ISSUE #6145: Shape Element Quick Menu
## Description

Adds quick action menu to shape layer when selected element is shape.
https://www.figma.com/file/1YM3jA9h4QGc4xLdyaXIJS/Stories-Sprint-1.7?node-id=7347%3A50908

Should have the following abilities: 
- Change background color
- Add animation
- Add link to element
- Clear media styles
- Insert media
- Insert text

Screen Shot 2021-02-01 at 10 55 47 AM

**Updated designs**
Screen Shot 2021-05-18 at 11 07 16 AM

## Acceptance Criteria 

- Clear element action should only appear if the element has styles applied. If the element doesn't have any styles applied the styles icon should be hidden. 
- Menu should have unit tests

### PR DESCRIPTION
**Pull request by embarks**
## Context
- Adding the quick actions for when a shape is selected.

## Summary
- updated test file (new unit tests for shape selection)
- now return shape actions in `useQuickActions`
- added a style panel state to the highlights context

## Relevant Technical Choices
- created a common action object to spread to other foreground selections
- added isPersistable prop to the link and animation panels so they are open when they are supposed to be highlighted and focused.

## User-facing changes

https://user-images.githubusercontent.com/41136059/119033056-a2d89000-b961-11eb-87c2-fa1cd9bea57d.mp4

### QA

- [x] This is a non-user-facing change and requires no QA

### UAT

- [x] UAT should use the same steps as above.

### Does this PR have a security-related impact?
no

### Does this PR change what data or activity we track or use?
no

### Does this PR have a legal-related impact?
no

## Checklist

- [x] This PR addresses an existing issue and I have linked this PR to it in ZenHub
- [x] I have tested this code to the best of my abilities
- [x] I have verified accessibility to the best of my abilities ([docs](https://github.com/google/web-stories-wp/blob/main/docs/accessibility-testing.md))
- [x] I have verified i18n and l10n (translation, right-to-left layout) to the best of my abilities
- [x] This PR contains automated tests (unit, integration, and/or e2e) to verify the code works as intended ([docs](https://github.com/google/web-stories-wp/tree/main/docs#testing))
- [x] I have added documentation where necessary
- [x] I have added a matching `Type: XYZ` label to the PR

---

Resolves #6145

### CODE REVIEW

**Comment by samwhale** [comment]
I think you fixed this issue too! https://github.com/google/web-stories-wp/issues/7624

File: assets/src/edit-story/app/highlights/quickActions/useQuickActions.js [hunk 1]
```diff
@@ -231,14 +234,8 @@ const useQuickActions = () => {
     handleMouseDown,
   ]);
 
-  const foregroundImageActions = useMemo(
+  const foregroundCommonActions = useMemo(
```
Comment (samwhale) [inline assets/src/edit-story/app/highlights/quickActions/useQuickActions.js:237]: :smart:

**Review by samwhale — APPROVED** [review]
This is dope! I think it would be dope to add karma tests for these too in this file: https://github.com/google/web-stories-wp/blob/main/assets/src/edit-story/components/canvas/karma/quickActions.karma.js

File: assets/src/edit-story/app/highlights/quickActions/useQuickActions.js (hunk 1 above)
Comment (BrittanyIRL) [reply assets/src/edit-story/app/highlights/quickActions/useQuickActions.js:237]: nice idea! 

let's keep an eye on this for ordering discontinuity.

**Review by BrittanyIRL — APPROVED** [review]
Looks great! 🔺  🟦  🟢 

(you do have some tests failing, but i know you'll fix those before merging 🙃  )

**Review by maxyinger — APPROVED** [review]
benissimo 👏

**Pull request merged**